### PR TITLE
Adds Replay Attack protection to the Wallet

### PR DESCRIPTION
**Pull request by alexvandesande**
Also refactors and simplifies token transfers

### CODE REVIEW

**Comment by alexvandesande** [comment]
I've notice a weird behaviour that I can't put my finger on. Simple DAO transfers are not working with for me in this PR. It's weird because calling the exact same function and parameters from the DAO directly will make a valid transaction, but via the send page it won't. I've tested with unicorns and they are working just fine.

screen-shot-2016-08-04-at-6 34 01-pm

The Token Transfer code is all ready, I just didn't enable it because it needs the user to click the approve button and that's not done yet.

**Comment by frozeman** [comment]
maybe the estimate gas is wrong? Or the node connection dropped?

**Comment by alexvandesande** [comment]
The gas estimate is the same on both. I'll check if an error is dropping the connection there.

**Comment by alexvandesande** [comment]
Found the previous bug, I was sending ether with the token transfer. Fixed it and it's working fine now.

**Comment by frozeman** [comment]
- [ ] Adding custom data doesn't work anymore on the send page.
E.g. if i select "more option" and then put in data into the data field, it doesn't pass that thought o the submit function. This also doesn't work, if i select a contract wallet.
- [ ] When going to the "Deploy contract" page, and select yet code and paste in a byte code, it doesn't use this yet code, but rather the one from the source code tab (the compiled source code)

**Comment by alexvandesande** [comment]
All solved. Ready to be merged.

File: app/client/templates/elements/compileContract.js [hunk 1]
```diff
@@ -59,10 +59,9 @@ Template['elements_compileContract'].onCreated(function() {
         var selectedToken = TemplateVar.getFrom('.select-token', 'selectedToken');
         var mainRecipient = TemplateVar.getFrom('div.dapp-address-input input.to', 'value');
         var replayProtectionOn = TemplateVar.get('replay-protection-checkbox');
+        var selectedType = TemplateVar.get('selectedType');
 
-        console.log('autorun', selectedContract, replayProtectionOn, selectedToken )
-
-        if(selectedContract){        
+        if(selectedType == 'source-code' && selectedContract){
```
Comment (frozeman) [inline app/client/templates/elements/compileContract.js]: this should be a tripple `===`, which means same content and type.
Its just good practice and would throw an error in ads lint

File: app/client/templates/elements/compileContract.js [hunk 2]
```diff
@@ -38,27 +47,66 @@ Template['elements_compileContract'].onCreated(function() {
...
+        
+        // selected contract
         var selectedContract = TemplateVar.get('selectedContract');
         var constructorInputs = _.clone(TemplateVar.get('constructorInputs'));
+        var selectedToken = TemplateVar.getFrom('.select-token', 'selectedToken');
+        var mainRecipient = TemplateVar.getFrom('div.dapp-address-input input.to', 'value');
+        var replayProtectionOn = TemplateVar.get('replay-protection-checkbox');
+        var selectedType = TemplateVar.get('selectedType');
+
+        if(selectedType == 'source-code' && selectedContract){
```
Comment (frozeman) [inline app/client/templates/elements/compileContract.js]: this should be a tripple ===, which means same content and type.
Its just good practice and would throw an error in ads lint

**Comment by frozeman** [comment]
Please pull the `ethereum:tools` on your side also

**Comment by frozeman** [comment]
@alexvandesande @luclu please test this PR again:

Please test the following:
- [ ] sending though the split with accounts
- [ ] with wallets
- [ ] with tokens
- [ ] and sending plain.

**Comment by frozeman** [comment]
Please pull the ethereum:tools on your side also

**Pull request merged**